openstf/pipeline/optimize_hyperparameters.py

```diff
@@ -68,11 +117,44 @@ def optimize_hyperparameters_pipeline(
     study = optuna.create_study(
         pruner=optuna.pruners.MedianPruner(n_warmup_steps=5), direction="minimize"
     )
+    model_type = pj["model"]
+    model = ModelCreator.create_model(model_type)
+
+    # Combine the default parameters with all model specific parameters
+    paramspace = default_paramspace.copy()
+    paramspace.update(**xgb_paramspace, **lgb_paramspace)
+
+    model_params = get_relevant_model_paramspace(model, paramspace)
+
+    if model_type == "lgb":
```

Comment:
Like my previous comment, this model specific stuff should probably be part of the objective.